src/MySqlConnector.Logging.log4net/Log4netLoggerProvider.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Globalization;
+using System.Reflection;
+using log4net;
+using log4net.Core;
+
+namespace MySqlConnector.Logging
+{
+    public sealed class Log4netLoggerProvider : IMySqlConnectorLoggerProvider
+    {
+#if NET45 || NET46
+		public IMySqlConnectorLogger CreateLogger(string name) => new Log4netLogger(LogManager.GetLogger(typeof(Log4netLogger).Assembly, "MySqlConnector." + name));
+#else
+		public IMySqlConnectorLogger CreateLogger(string name) => new Log4netLogger(LogManager.GetLogger(typeof(Log4netLogger).GetTypeInfo().Assembly, "MySqlConnector." + name));
```

Comment:
Does this not work on `NET45 || NET46`? Or is this an optimization?

Would it be worth caching the assembly in a static?

Reply:
Good call; that code path works for both. I'll add a `static`.